src/JSONFileReader.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "Board.hpp"
+
+using json = nlohmann::json;
+
+class JSONFileReader
+{
+    json j_;
+public:
+    explicit JSONFileReader(std::string path);
+    int readRowsNumber();
+    int readColsNumber();
+    Lines readRows();
+    Lines readCols();
+};
+
+class InvalidDimensions : public std::exception
+{
+    std::string reason;
+public:
+    explicit InvalidDimensions(const char* why)
+        : reason(why)
+    {}
+
+    virtual const char* what() const throw()
```

Comment:
throw() specifier is deprecated from C++11 and is removed in C++17. Do not use it.